client/my-sites/plans-grid/components/billing-timeframe.tsx

```diff
@@ -9,11 +9,14 @@ import {
 	TERM_ANNUALLY,
 	isWooExpressPlan,
 	PLAN_HOSTING_TRIAL_MONTHLY,
+	isFreePlan,
 } from '@automattic/calypso-products';
 import { formatCurrency } from '@automattic/format-currency';
+import { isAnyHostingFlow } from '@automattic/onboarding';
```

Comment:
I need to have a look here. `flowname` shouldn't leak further into plans-grid as that's a concept we are removing.

Please wait for explicit approval from code owners before merging anything with this PR (and ping if we are not timely).

Reply:
It's OK, we can move to `intent`. It's just a bad habit that dies hard :)

Reply:
Thanks for giving me a chance to respond 😬 

To me, both `flowName` and `intent` bring similar context into the `plans-grid`. `intent` is stronger in that it can apply to both admin and signup (as the case with Newsletter), and having a single concept throughout is likely to lead to less complexity. However, it's still context that we ought to try and minimize its use/effect, or perhaps eliminate altogether from `plans-grid` UI components. 

We've been trying to separate data concerns from UI and that's something we ought to keep doing. I think this case may suffice for a different control/design with that in mind. 

So instead of putting together the refund period here (it very much sounds like a "pricing meta" to me) and conditioning with:

```
	if ( ! isAnyHostingFlow( flowName ) || isFreePlan( planSlug ) ) {
		return null;
	}
```

I would propose the following:

1. Compute the refund period in `usePricingMetaForGridPlans` and include as a property of `PricingMeta` e.g. `refundPeriod?: number`. right [here](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plans-features-main/hooks/data-store/use-pricing-meta-for-grid-plans.ts#L207-L214), so something like:

```
{
    originalPrice: planPrices[ planSlug ]?.originalPrice,
    discountedPrice: planPrices[ planSlug ]?.discountedPrice,
    billingPeriod: pricedAPIPlan?.bill_period,
    currencyCode: pricedAPIPlan?.currency_code,
    introOffer: sitePlan?.introOffer,
    expiry: sitePlan?.expiry,
    ...( pricedAPIPlan?.bill_period && ! isFreePlan( planSlug ) && { refundPeriod: pricedAPIPlan?.bill_period === 365 : 14 : 7 } )
}
```

2. Add an optional property to [PlansGridProps](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plans-grid/index.tsx#L31) to show/hide the refund notice e.g. `showRefundNoticeInBillingDescriptions` (or something less verbose?), which can condition on the flow or intent that we want the notice to be shown (no need for `isFreePlan` check):

```
showRefundNoticeInBillingDescriptions: isAnyHostingFlow( flowName )
```

I think that would be a good separation of concerns. Let me know what you think. I can also do it in a follow-up if you want to ship earlier.

Reply:
> Let me know what you think

Double this! 🙇 I could be wrong, always. Just feels to me there are multiple concerns